src/Sarif.UnitTests/SarifVersionOneToCurrentTests.cs

```diff
@@ -1453,5 +1483,336 @@ public void SarifTransformerTests_ToCurrent_TwoResultsWithFixes()
+          ""kind"": ""type""
+        },
+        ""collections"": {
+          ""kind"": ""namespace""
+        }
+      },
+      ""results"": [
+        {
+          ""ruleId"": ""C2001"",
+          ""message"": {
+            ""arguments"": [
+              ""ptr""
+            ]
+          },
+          ""ruleMessageId"": ""default"",
+          ""analysisTarget"": {
+            ""uri"": ""file:///home/buildAgent/src/collections/list.cpp""
+          },
+          ""locations"": [
+            {
+              ""physicalLocation"": {
+                ""fileLocation"": {
+                  ""uri"": ""file:///home/buildAgent/src/collections/list.h""
+                },
+                ""region"": {
+                  ""startLine"": 1,
+                  ""startColumn"": 1,
+                  ""endLine"": 1,
+                  ""endColumn"": 28,
+                  ""length"": 27,
+                  ""snippet"": {
+                    ""text"": ""add_core(ptr, offset, val);""
+                  }
+                }
+              },
+              ""fullyQualifiedLogicalName"": ""collections::list::add""
+            }
+          ],
+          ""codeFlows"": [
+            {
+              ""message"": {
+                ""text"": ""Path from declaration to usage""
+              },
+              ""threadFlows"": [
+                {
+                  ""locations"": [
+                    {
+                      ""step"": 1,
+                      ""location"": {
+                        ""physicalLocation"": {
+                          ""fileLocation"": {
+                            ""uri"": ""file:///home/buildAgent/src/collections/list.h""
+                          },
+                          ""region"": {
+                            ""startLine"": 15,
+                            ""snippet"": {
+                              ""text"": ""int *ptr;""
+                            }
+                          }
+                        },
+                        ""fullyQualifiedLogicalName"": ""collections::list:add"",
+                        ""message"": {
+                          ""text"": ""Variable \""ptr\"" declared.""
+                        }
+                      },
+                      ""kind"": ""Declaration"",
+                      ""module"": ""platform"",
+                      ""importance"": ""essential"",
+                      ""properties"": {
+                        ""sarifv1/annotations"": null,
+                        ""sarifv1/essential"": false,
+                        ""sarifv1/fullyQualifiedLogicalName"": ""collections::list:add"",
+                        ""sarifv1/id"": 0,
+                        ""sarifv1/logicalLocationKey"": null,
+                        ""sarifv1/taintKind"": 0,
+                        ""sarifv1/target"": null,
+                        ""sarifv1/targetKey"": null,
+                        ""sarifv1/threadId"": 52,
+                        ""sarifv1/values"": null
```

Comment:
Bug: You need to bag v1  `annotatedCodeLocation.state` and `targetLocation` as well.

Until Michael lets us off the hook 😄.

Reply:
Done

Reply:
State transfers to v2

Reply:
No v1 targetLocation property

Reply:
Ah right. We need a test for that.

Reply:
The "ah right, we need a test" was about `state`.

As to `targetLocation`, [are you sure](https://rawgit.com/sarif-standard/sarif-spec/master/Static%20Analysis%20Results%20Interchange%20Format%20(SARIF).html#annotatedcodelocation-target)?

Reply:
[Yes](https://github.com/oasis-tcs/sarif-spec/raw/master/Documents/WorkingDrafts/sarif-v1.0-wd01.docx)
[Schema](https://github.com/Microsoft/sarif-sdk/blob/sarif-v2/src/Sarif/Schemata/Sarif-1.0.0.schema.json)

Reply:
Well, ok, that was a bug in the v1 schema. It didn't implement the v1 spec.

Reply:
Done